Mathlib/Data/Fintype/Order.lean

```diff
@@ -152,6 +152,16 @@ noncomputable abbrev toCompleteLinearOrderOfNonempty [LinearOrder α] : Complete
 
 end Nonempty
 
+lemma exists_le_maximal {α} [Fintype α] [PartialOrder α] {a : α} {p : α → Prop}
```

Comment:
```
lemma Finite.exists_ge_minimal {α} [Finite α] [PartialOrder α] {a : α} {p : α → Prop} (h : p a) :
    ∃ b, b ≤ a ∧ Minimal p b := by
  obtain ⟨b, ⟨hba, hb⟩, hbmin⟩ :=
    Set.Finite.exists_minimal_wrt id {x | x ≤ a ∧ p x} (Set.toFinite _) ⟨a, rfl.le, h⟩
  exact ⟨b, hba, hb, fun x hx hxb ↦ (hbmin x ⟨hxb.trans hba, hx⟩ hxb).le⟩

lemma Finite.exists_le_maximal {α} [Finite α] [PartialOrder α] {a : α} {p : α → Prop} (h : p a) :
    ∃ b, a ≤ b ∧ Maximal p b :=
  Finite.exists_ge_minimal (α := αᵒᵈ) h 
```

Reply:
I think that `Finite` makes more sense here than `Fintype`. See [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Lattice.html#Finite.exists_max), for instance.

Reply:
Fixed:)